src/indexed.js

```diff
@@ -0,0 +1,32 @@
+import {Seq} from 'immutable';
+import Base from './base';
+import {updateCursor} from './utils';
+
+function Indexed(root, data, keyPath, onChange, size) {
```

Comment:
May be clearer to use the `class` syntax here...? Unless you're about to rewrite this in a completely functional way.